### ISSUE TITLE
Endpoint struct_url missing URI authority key

### ISSUE THREAD

**Issue by jowrjowr** [phoenixframework/phoenix#4739]
### Environment

* Elixir version (elixir -v):

```
Erlang/OTP 24 [erts-12.2] [source] [64-bit] [smp:32:32] [ds:32:32:10] [async-threads:1] [jit]

Elixir 1.12.3 (compiled with Erlang/OTP 24)
```

* Phoenix version (mix deps):

```
mix deps | grep phoenix
* opentelemetry_phoenix 1.0.0-rc.7 (Hex package) (mix)
  locked at 1.0.0-rc.7 (opentelemetry_phoenix) 233a02e1
* phoenix 1.6.6 (Hex package) (mix)
  locked at 1.6.6 (phoenix) 807bd646
* phoenix_ecto 4.4.0 (Hex package) (mix)
  locked at 4.4.0 (phoenix_ecto) 09864e55
* phoenix_html 3.2.0 (Hex package) (mix)
  locked at 3.2.0 (phoenix_html) 36ec97ba
* phoenix_live_dashboard 0.6.5 (Hex package) (mix)
  locked at 0.6.5 (phoenix_live_dashboard) ef4fa50d
* phoenix_live_reload 1.3.3 (Hex package) (mix)
  locked at 1.3.3 (phoenix_live_reload) 76679667
* phoenix_live_session 0.1.3 (Hex package) (mix)
  locked at 0.1.3 (phoenix_live_session) ae353587
* phoenix_live_view 0.17.7 (Hex package) (mix)
  locked at 0.17.7 (phoenix_live_view) 25eaf410
* phoenix_pubsub 2.0.0 (Hex package) (mix)
  locked at 2.0.0 (phoenix_pubsub) c52d948c
* phoenix_view 1.1.2 (Hex package) (mix)
  locked at 1.1.2 (phoenix_view) 7ae90ad2
```

* Operating system:

Linux

### Expected behavior

I expect that using the URI struct generated by `Endpoint.struct_url/0` to be able to used as such, eg:

```
Endpoint.struct_url |> URI.merge("/some/other/stuff")
```

generates 

```
%URI{
  authority: "localhost:4000",
  fragment: nil,
  host: "localhost",
  path: "/some/other/stuff",
  port: 4000,
  query: nil,
  scheme: "http",
  userinfo: nil
}
```

### Actual behavior

Kaboom.

```
ElixirApiWeb.Endpoint.struct_url() |> URI.merge("/some/other/stuff")
** (ArgumentError) you must merge onto an absolute URI
    (elixir 1.12.3) lib/uri.ex:666: URI.merge/2
```

The issue is that the URI struct generated is lacking the authority key.

Compare and contrast.

```
iex(18)> ElixirApiWeb.Endpoint.static_url() 
"http://localhost:4000"
iex(19)> ElixirApiWeb.Endpoint.static_url() |> URI.parse()
%URI{
  authority: "localhost:4000",
  fragment: nil,
  host: "localhost",
  path: nil,
  port: 4000,
  query: nil,
  scheme: "http",
  userinfo: nil
}
iex(20)> ElixirApiWeb.Endpoint.struct_url()
%URI{
  authority: nil,
  fragment: nil,
  host: "localhost",
  path: nil,
  port: 4000,
  query: nil,
  scheme: "http",
  userinfo: nil
}
iex(21)>
```

This appears to be an example of a previously fixed bug as found previously: https://github.com/phoenixframework/phoenix_live_view/pull/1548

**Comment by Gazler** [comment]
I believe this is fixed in Elixir 1.13 in this commit:

https://github.com/elixir-lang/elixir/commit/756611099d2ece3e4ec7c242ed4d7ea5838b246a

**Comment by leandrocp** [comment]
On Elixir v1.13.0 I get:

```
iex> DemoWeb.Endpoint.struct_url() |> URI.merge("/some/other/stuff")
%URI{
  authority: nil,
  fragment: nil,
  host: "localhost",
  path: "/some/other/stuff",
  port: 4000,
  query: nil,
  scheme: "http",
  userinfo: nil
}
```

```
❯ elixir --version
Erlang/OTP 24 [erts-12.0.3] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit]

Elixir 1.13.0 (compiled with Erlang/OTP 24)
```
